### ISSUE TITLE
Fix escaping of special characters in tag names

### ISSUE THREAD

**Issue by roundhill** [Automattic/simplenote-electron#3279]
Please see https://github.com/Automattic/simplenote-android/issues/1394 over at the Android repo. I _think_ the web app is likely the cause, but it could be Android as well.

**To Reproduce**
* Create a note with a tag on it with a dash, like `test-ing`.
* Head over to the web and create another note, and use the `test-ing` tag. I did the tab autocomplete to add the tag.
* You should now have two `test-ing` tags, but one of them is actually using a url escaped dash `%2D`.

**Comment by codebykat** [comment]
@roundhill I have tried and tried but cannot seem to reproduce this with only the web app. I'd love to get to the bottom of it as it's been reported sporadically for years, but I'm not seeing it 😭

 I have created tags with hyphens as well as a TON of other special characters. In your steps, did you create the initial tag on Android? Are we sure it isn't the Android app incorrectly escaping the tag name? Where are you seeing the screenshot of the incorrect tag name, was that the tag list on the web?

I inspected the tag in Simperium to verify it isn't getting encoded:

![Image](https://github.com/user-attachments/assets/797c850c-167d-4c54-ad89-580da9e73b11)

As a side note, it's clear from my tags on my testing account that I've been trying to reproduce this bug for some years.

![Image](https://github.com/user-attachments/assets/1890643a-78bc-4529-b4f0-44efc01aff3f)

**Comment by roundhill** [comment]
Yeah, my bad there. The first step should be to create a note on another client. Here's what I just tested:

1. Create a new note on iPhone with a new tag: `jan-15`.
2. Switch to web app, observe note and tag are there.
3. Create another note, add the tag by starting to type `jan` and then tab to autocomplete.
4. You should now have two `jan-15` tags in your tags list.

**Comment by roundhill** [comment]
Maybe the bug is on iOS or Android, this is after creating a tag on iOS without messing with the web app 🤔 

<img width="322" alt="Image" src="https://github.com/user-attachments/assets/c573e10e-d2fe-4811-bf15-68ed0a780ff6" />

**Comment by roundhill** [comment]
Ok @jleandroperez informed me of this effort to escape the simperium key: https://github.com/Automattic/simplenote-ios/pull/828


It looks like @sandymcfadden started on it here but it wasn't merged:
https://github.com/Automattic/simplenote-electron/pull/3017

We need to ensure that the Simperium key gets percent-escaped in the same way that the other clients are doing it.
